### DISCUSSION TITLE
PromptConfig - @inquirer/prompts v3 to @inquirer/prompts v4

### DISCUSSION THREAD

**Discussion by tuler** [SBoudrias/Inquirer.js#1362] [Q&A] [answered]
I'm migrating some code from `@inquirer/prompts@3.0` to `@inquirer/prompts@4.1`.
I noticed the `PromptConfig` type is not exported anymore, and `input` now receives a `InputConfig`, which is not exported either.

I used `PromptConfig` to create simple "custom inputs", more like utility functions, that call `input` and do some extra goodies, like validation, or typing. Something like this:

```typescript
import { PromptConfig, input } from "@inquirer/prompts";
import { Address, isAddress, getAddress } from "viem";

/**
 * Prompt for an address value.
 * @param config inquirer config
 * @returns address
 */
export type AddressPromptConfig = PromptConfig<{ default?: Address }>;
export const addressInput = async (
    config: AddressPromptConfig,
): Promise<Address> => {
    const address = await input({
        ...config,
        validate: (value) => isAddress(value) || "Enter a valid address",
    });
    return getAddress(address);
};
```

What is the recommended way of doing this kind of stuff now, without having to write a full blown custom prompt like the procedure described at the [@inquirer/core documentation](https://github.com/SBoudrias/Inquirer.js/tree/master/packages/core)?

**Comment by SBoudrias** [comment, marked as answer]
Hi, the `PromptConfig` just disappeared - it's not used anymore.

To get the input config type, I would recommend going at it this way:

```ts
type InputConfig = Parameters<typeof input>[0];

export type AddressPromptConfig = InputConfig & { default?: Address };
```
